File: 387_First_Unique_Character_in_a_String/firstUniqChar.cpp
#include <iostream>
#include <unordered_map> 

using namespace std;

class Solution {
	public:
    	int firstUniqChar(string s) {
            unordered_map<char, int> counts;
            for(int i = 0; i < s.size(); i++)
                counts[s[i]]++;
            for(int i = 0; i < s.size(); i++)
                if(counts[s[i]] == 1)
                    return i;
            return -1;
    	}
};

File: 387_First_Unique_Character_in_a_String/firstUniqChar.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        counts = {}
        for c in s:
            if c not in counts:
                counts[c] = 1
            else:
                counts[c] += 1

        for idx, c in enumerate(s):
            if counts[c] == 1:
                return idx
        return -1

sol = Solution()

